### ISSUE TITLE
pulumi/aws v5.35.0 breaks cognito UserPool tags

### ISSUE THREAD

**Issue by shawnjones253** [pulumi/pulumi-aws#2533]
### What happened?

Versions of `@pulumi/aws` > `5.35.0` cause `tags` not to be applied when creating a new `aws.cognito.UserPool`.

Reverting to `5.34.0` fixes the issue.

### Expected Behavior

UserPool tags should be applied.

### Steps to reproduce

```
mkdir example && cd example && pulumi new aws-typescript --name myproject
npm install @pulumi/aws@5.35.0 -E
```

```TypeScript
// index.ts
import * as aws from "@pulumi/aws";

const userPool = new aws.cognito.UserPool("cognito-user-pool", {
  name: "test-cognito-user-pool",
  tags: {
    tagKey: "tagValue",
  },
});

export const poolId = userPool.id;
```

```
pulumi up --yes
```

Look at the tags of the new UserPool in AWS console, note they are missing.

```
pulumi destroy --yes
npm install @pulumi/aws@5.34.0 -E
pulumi up --yes
```

Tags will be present on the UserPool in AWS console.

### Output of `pulumi about`

```
Backend
Name           pulumi.com
URL            https://app.pulumi.com/***
User           ***
Organizations  ***, ***

Dependencies:
NAME            VERSION
@pulumi/aws     5.35.0
@pulumi/awsx    1.0.2
@pulumi/pulumi  3.68.0
@types/node     16.18.32
```

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by mikhailshilkov** [comment]
Hi @shawnjones253 thank you for your issue.

I found the following:

- I can reproduce the bug in 5.36.0 of the provider, and also in the latest 5.41.0
- It actually works fine in 5.35.0 for me (could you double check?)
- I don't get the same issue in TF with provider versions 4.61, 4.62, or 4.65
- Pulumi versions point me to [this commit](https://github.com/pulumi/terraform-provider-aws/commit/9b8094e6b35520bcc65f894dd530db4d6f6bd362)
- I'm not yet sure what's wrong about it that only affects Pulumi but not TF

**Comment by rquitales** [comment]
It appears that the framework conversions were reverted in the upstream fork, specifically in [this commit](https://github.com/pulumi/terraform-provider-aws/commit/bbeec79a85191c9af92b0ef2b2c51d6515d09873). This change also affects the Cognito user pool resource.

On the other hand, the commit shared by @mikhailshilkov [here](https://github.com/pulumi/terraform-provider-aws/commit/9b8094e6b35520bcc65f894dd530db4d6f6bd362) extends the framework conversion work to tag logic by consolidating boilerplate code. To address this issue, a possible solution would be to revert the changes in this commit as well, similar to what we did with the framework conversion logic.

**Comment by shawnjones253** [comment]
any update here? hoping for something that will "definitely fix" rather than "may fix" -- and hopefully something sooner than the 6.0 release

**Comment by mikhailshilkov** [comment]
Thank you for your patience @shawnjones253

AWS v6 will support TF plugin framework resources first-class, so the buggy workaround has been removed there. I tested your repro with HEAD of v6 and it is indeed fixed - I can see the tagKey/tagValue tag in the console.

We are finalising the last tasks for 6.0 release and will launch it as soon as possible, hopefully later this week.

I'll go ahead and close the issue as fixed, marking it with a 6.0 label.
